scvelo/tools/dynamics/_dynamics_base.py

```diff
@@ -0,0 +1,768 @@
+                    test_bimodality(weighted_counts[:, col_id], kde=True)
+                    for col_id in range(counts.shape[1])
+                ]
+            )
+            means = np.array(means).T
+        except ValueError:
+            pvals = np.ones(counts.shape[1])
+            means = np.zeros((counts.shape[1], counts.shape[1]))
+
+        self.pval_steady = max(pvals)
+        self.steady_state = means[1, :]
+
+        self._initialize_parameters(weighted_counts)
+
+        model_params = self.get_model_parameters()
+        self.t_ = self.tau_inv(state=self.initial_state_[None, :], **model_params)
+
+        model_params.update({"t_": self.t_, "scaling": self.scaling[0]})
+        self.params = pd.DataFrame(model_params, index=[0])
+
+        self.t, self.tau, self.o = self.get_time_assignment()
+
+        self.loss_ = [self.get_loss()]
+
+        for kwargs in initial_parameter_fit:
+            self.initial_parameter_fit(**kwargs)
+
+        self._set_steady_state_ratio(**self.get_model_parameters())
+
+        return self
+
+    # TODO: Add case `weighted == "outer"`
+    def get_weights(self, weighted=False, weights_cluster=None):
+        if not weighted:
+            weights = np.ones(len(self.weights_), bool)
+        elif weighted == "upper":
+            weights = self.weights_upper_
+        else:
+            weights = self.weights_
+
+        if weights_cluster is not None and len(weights) == len(weights_cluster):
+            weights &= weights_cluster
+        return weights
+
+    # TODO: Try to remove function
+    def get_counts(self, scaling=None, weighted=False, weights_cluster=None):
+        scaling = self.scaling if scaling is None else scaling
+
+        counts = self.raw_counts / scaling
+
+        if weighted or weights_cluster is not None:
+            weights = self.get_weights(
+                weighted=weighted, weights_cluster=weights_cluster
+            )
+            counts = counts[weights, :]
+
+        return counts
+
+    def get_vars(self, scaling=None, t_=None, initial_state_=None, **model_parameters):
+        scaling = self.scaling if scaling is None else scaling
+        if t_ is None or t_ == 0:
+            t_ = (
+                self.t_
+                if initial_state_ is None
+                else self.tau_inv(
+                    state=initial_state_,
+                    **self.get_model_parameters(**model_parameters),
+                )
+            )
+        return scaling, t_
+
+    def get_model_parameters(self, after_switch=False, **kwargs):
+        model_parameters = {
+            model_parameter: kwargs.get(model_parameter, getattr(self, model_parameter))
+            for model_parameter in self.model_parameters
+        }
+
+        if after_switch:
+            self._set_parameters_after_switch(model_parameters)
+
+        return model_parameters
+
+    @abstractmethod
+    def _set_parameters_after_switch(self, model_parameters):
+        return model_parameters
```

Comment:
Yes, because I do not know beforehand which model parameters have to be changed after the switch. This method will have to be implemented in the child class - for example [here](https://github.com/WeilerP/scvelo/blob/d71a3d4f2a45225e622aeb968849ebaa91542abb/scvelo/tools/dynamics/_recovery.py#L109) for the splicing model.